Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.cs

```diff
@@ -366,6 +375,15 @@ protected void MuzzleFlash(EntityUid gun, AmmoComponent component, EntityUid? us
         CreateEffect(gun, ev, user);
     }
 
+    public void CauseImpulse(EntityCoordinates fromCoordinates, EntityCoordinates toCoordinates, PhysicsComponent userPhysics)
+    {
+        var fromMap = fromCoordinates.ToMap(EntityManager);
+        var toMap = toCoordinates.ToMapPos(EntityManager);
+        var mapDirection = toMap - fromMap.Position;
+
+        var impulseVector =  mapDirection.Normalized * userPhysics.Mass;
+        Physics.ApplyLinearImpulse(userPhysics, -impulseVector * 1.2f);
```

Comment:
not sure about this calculation of the impulse. It's the shooters mass times 1.2?

I dont know if different people/mobs have different mass in-game, and this might still not be the route that should be taken, but wouldnt just a set/constant impulse strength be more reasonable? in the future we might want to make it reliant on the gun type/bullet as well

```suggestion
        const float impulseStrength = 85.0f; //The bullet impulse strength
        var impulseVector =  mapDirection.Normalized * impulseStrength;
        Physics.ApplyLinearImpulse(userPhysics, -impulseVector);
```

(the 85f is based on the standard body mass of ~71 * 1.2)

Reply:
alright everything is done